### PR TITLE
MDS: Fixup for Sparse Data

### PR DESCRIPTION
**Pull request by jerneju**
##### Issues
* Scatter Plot Graph cannot handle sparse data.
* As of #2648 MDS did not work for sparse data.

##### Description of changes
* Scatter Plot Graph and Scaling can now handle sparse data and primitive data in metas by themselves.
* Fix MDS to handle sparse data.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 1]
```diff
@@ -691,7 +720,7 @@ def update_data(self, attr_x, attr_y, reset_view=True):
             self.plot_widget.addItem(self.density_img)
 
         self.data_indices = np.flatnonzero(self.valid_data)
-        if len(self.data_indices) != self.original_data.shape[1]:
+        if len(self.data_indices) != len(self.original_data):
```
Comment (nikicc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: Why this change?

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 2]
```diff
@@ -759,21 +788,15 @@ def set_labels(self, axis, labels):
...
-        size_index = self.get_size_index()
-        if size_index == -1:
+        if self.attr_size is None:
             size_data = np.full((self.n_points,), self.point_width,
                                 dtype=float)
         else:
             size_data = \
                 self.MinShapeSize + \
-                self.scaled_data[size_index, self.valid_data] * \
+                self.scaled_data.get_column_view(self.attr_size)[0][self.valid_data] * \
```
Comment (nikicc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: Is this needed if we densified before?

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 3]
```diff
@@ -782,7 +811,7 @@ def compute_sizes(self):
         return size_data
 
     def update_sizes(self):
-        self.master.prepare_data()
+        self.sparse_to_dense()
```
Comment (nikicc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: Don't we also need the `self.set_data` call here? Same for a few lines below.

Comment (jerneju) [reply Orange/widgets/visualize/owscatterplotgraph.py]: Yes, after `self.set_data(self.sparse_to_dense())` .

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/367233902/
